GrayBMP.cs

```diff
@@ -126,6 +125,39 @@ public void Dirty ()
       End ();
    }
 
+   /// <summary>
+   /// Draws a thick line between the two given end-points (with given size & given shade of gray)
+   /// </summary>
+
+   public void DrawThickLine (int x1, int y1, int x2, int y2, int width, int color) {
+      Begin (); 
+      Check (x1, y1); Check (x2, y2); Dirty (x1, y1, x2, y2);
+
+      const double OneRadian = PI / 180;
+      double offset = width * 0.5;
+      var BaseAngle = Atan2 (y2 - y1, x2 - x1) / OneRadian;
+
+      System.Drawing.Point[] HexPoints = new System.Drawing.Point[8];

```

Comment:
```suggestion
      var hexPts = new System.Drawing.Point[8];

```

Reply:
Instead of using a local array, as this is called multiple times, you can use list and reuse when drawing thick line.

Reply:
In list, it seems we can't assign a value for (i+1)th index, without assigning for ith index, even though we declare the size.  So, I used array instead of list.